src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,7 +149,7 @@ public function loadChoicesForValues(array $values, $value = null)
 
         // Optimize performance in case we have an object loader and
         // a single-field identifier
-        $optimize = null === $value || is_array($value) && $value[0] === $this->idReader;
+        $optimize = is_array($value) && $this->idReader && $value[0];
```

Comment:
the previous check was right. Here, you check that ``$value[0]`` is any truthy value, which does not make any sense (it will always be the case for a callable array anyway)